pkg/cloudprovider/aws/fleet/packing/instancetype.go

```diff
@@ -0,0 +1,159 @@
+	instanceTypesAvailable = []*instanceType{
+		{
+			name: "m5.8xlarge",
+			totalCapacity: v1.ResourceList{
+				v1.ResourceCPU:    resource.MustParse("32000m"),
+				v1.ResourceMemory: resource.MustParse("128Gi"),
+			},
+			availableCapacity: v1.ResourceList{
+				v1.ResourceCPU:    resource.MustParse("32000m"),
+				v1.ResourceMemory: resource.MustParse("128Gi"),
+			},
+			pricePerHour: 1.536,
+		},
+		// {
+		// 	name: "m5.4xlarge	",
+		// 	totalCapacity: v1.ResourceList{
+		// 		v1.ResourceCPU:    resource.MustParse("16000m"),
+		// 		v1.ResourceMemory: resource.MustParse("64Gi"),
+		// 	},
+		// 	availableCapacity: v1.ResourceList{
+		// 		v1.ResourceCPU:    resource.MustParse("16000m"),
+		// 		v1.ResourceMemory: resource.MustParse("64Gi"),
+		// 	},
+		// 	pricePerHour: 0.768,
+		// },
+		{
+			name: "m5.2xlarge",
+			totalCapacity: v1.ResourceList{
+				v1.ResourceCPU:    resource.MustParse("8000m"),
+				v1.ResourceMemory: resource.MustParse("32Gi"),
+			},
+			availableCapacity: v1.ResourceList{
+				v1.ResourceCPU:    resource.MustParse("8000m"),
+				v1.ResourceMemory: resource.MustParse("32Gi"),
+			},
+			pricePerHour: 0.384,
+		},
+		{
+			name: "m5.xlarge",
+			totalCapacity: v1.ResourceList{
+				v1.ResourceCPU:    resource.MustParse("4000m"),
+				v1.ResourceMemory: resource.MustParse("16Gi"),
+			},
+			availableCapacity: v1.ResourceList{
+				v1.ResourceCPU:    resource.MustParse("4000m"),
+				v1.ResourceMemory: resource.MustParse("16Gi"),
+			},
+			pricePerHour: 0.192,
+		},
+		{
+			name: "m5.large",
+			totalCapacity: v1.ResourceList{
+				v1.ResourceCPU:    resource.MustParse("2"),
+				v1.ResourceMemory: resource.MustParse("8Gi"),
+			},
+			availableCapacity: v1.ResourceList{
+				v1.ResourceCPU:    resource.MustParse("2"),
+				v1.ResourceMemory: resource.MustParse("8Gi"),
+			},
+			pricePerHour: 0.096,
+		},
+	}
+)
+
+type instanceType struct {
+	name string
+	sync.RWMutex
+	availableCapacity v1.ResourceList
+	totalCapacity     v1.ResourceList
+	pricePerHour      float64
+}
+
+func (p *PodPacker) getInstanceTypes(filter string) []*instanceType {
+	instanceTypeOptions := instanceTypesAvailable
+	// sort by increasing capacity of the instance
+	sort.Sort(byCapacity{instanceTypeOptions})
+	return instanceTypeOptions
+}
+
+func (r *instanceType) isAllocatable(cpu, memory resource.Quantity) bool {
+	r.Lock()
+	defer r.Unlock()
+	return (r.availableCapacity.Cpu().Cmp(cpu) >= 0) &&
+		(r.availableCapacity.Memory().Cmp(memory) >= 0)
+}
+
+func (r *instanceType) reserveCapacity(cpu, memory resource.Quantity) error {
+	r.Lock()
+	defer r.Unlock()
+	if r.availableCapacity.Cpu().IsZero() {
+		return fmt.Errorf("insufficient CPU capacity")
+	}
+	if r.availableCapacity.Memory().IsZero() {
+		return fmt.Errorf("insufficient Memory capacity")
+	}
+	r.availableCapacity.Cpu().Sub(cpu)
+	r.availableCapacity.Memory().Sub(memory)
+	return nil
+}
+
+func filterInstancesBasedOnCost(instanceTypeOptions []*instanceType) []*instanceType {
```

Comment:
is this filter or sort? 

I think just using 
```
	sort.Sort(byCost{instanceTypeOptions})
```
inline is pretty idiomatic. You might be able to cut this function entirely.